tests/Abuse/AbuseRedisTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Utopia\Tests;
+
+use DateInterval;
+use PHPUnit\Framework\TestCase;
+use Utopia\Abuse\Abuse;
+use Utopia\Abuse\Adapters\Redis;
+use Redis as Client;
+use Utopia\Exception;
+
+class AbuseRedisTest extends TestCase
```

Comment:
We should have the same thing as we have in databases where we have a base test that is extended and only the instance is initialized with different adapters.

Reply:
Done, had to refactor adapter as well